astropy/table/tests/test_operations.py

```diff
@@ -641,6 +641,18 @@ def test_stack_rows(self, operation_table_type):
                                  '1.0 bar',
                                  '1.0 bar']
 
+    def test_stack_columns(self, operation_table_type):
```

Comment:
For testing, you should include stuff like:
```
c = table.Column([1, 2])
mc = table.MaskedColumn([1, 2])
q = [1, 2] * u.m

t = table.hstack([c, q])
assert t.__class__ is table.QTable
assert t.masked is False

# Same for [mc, q] where out class is QTable and masked is True
# Same for [c, mc] where out class is Table and masked is True

with pytest.raises(ValueError):
    table.vstack([c, q])

t = table.vstack([q, q])
assert t.__class__ is table.QTable

t = table.vstack([c, c])
assert t.__class__ is table.Table

Reply:
I had rather keep these tests, too (more consistent how we tested the Row stacking) and added a bunch other ones you suggested here, too.